### PR TITLE
Added ParsingFetcher #275 that wraps a Raw type Parser and and Raw type Fetcher

### PR DESCRIPTION
**Pull request by magillus**
Reading and parsing network/file fetch can be done at same step with single Fetcher.

### CODE REVIEW

**Comment by magillus** [comment]
Enhancement from the `issue` #275

**Comment by stoyicker** [comment]
note to self: if this is merged, add to kotlin artifact

File: store/src/main/java/com/nytimes/android/external/store/base/impl/ParsingFetcher.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     * @param rawFetcher   fetches raw data by key
+     * @param parsedParser parses raw data to the instance of 'Parsed'
+     */
+    public ParsingFetcher(@Nonnull Fetcher<Raw, Key> rawFetcher, @Nonnull Parser<Raw, Parsed> parsedParser) {
+        this.rawFetcher = rawFetcher;
+        this.parser = parsedParser;
+    }
+
+    /**
+     * Returns parsing fetcher for fetcher and parser
```
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/ParsingFetcher.java]: This comment seems odd, collapse it to just the sentence and remove the auto gen parameter comments.

**Review by brianPlummer — APPROVED** [review]

File: store/src/main/java/com/nytimes/android/external/store/base/impl/ParsingFetcher.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * Creates ParsingFetcher for raw data type Fetcher and Raw data Parser.
+     */
+    public static final <Parsed, Raw, Key> ParsingFetcher<Parsed, Raw, Key> from(@Nonnull Fetcher<Raw, Key> fetcher, @Nonnull Parser<Raw, Parsed> parser) {
+        return new ParsingFetcher(fetcher, parser);
+    }
+
+    @Nonnull
+    @Override
+    public Observable<Parsed> fetch(@Nonnull Key key) {
```
Comment (ramonaharrison) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/ParsingFetcher.java:40]: Should be based off of the `feature/rx2` branch and return a `io.reactivex.Single`

**Review by ramonaharrison — CHANGES_REQUESTED** [review]
The `develop` branch is deprecated, we're not planning any future Store releases from this branch. Please rebase off of `feature/rx2` and update to return an RxJava2 `io.reactivex.Single`.

Otherwise, LGTM. Thanks!

**Comment by magillus** [comment]
I will close this PR because of different branch from my fork.

**Comment by pavlospt** [comment]
@magillus You can rebase it on `feature/rx2` instead of closing it.

**Comment by magillus** [comment]
There were some merge conflicts etc. my PR is simple adding a file and unit test so was faster just create branch from `feature/rx2`

**Pull request closed without merging**